src/Configuration/Properties/PropertyType.php

```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace MyParcelCom\Integration\Configuration\Properties;
+
+enum PropertyType: string
+{
+    case STRING = 'string';
+    case NUMBER = 'number';
+    case BOOLEAN = 'boolean';
+}
```

Comment:
My suggestion for the different field classes above does not dismiss the need of this enum. The enum is still important for rendering the json-schema in the response, while the field types becomes something else (since they go beyond the json schema property types)